packages/flutter_tools/test/src/context.dart

```diff
@@ -222,7 +222,9 @@ class FakeDeviceManager implements DeviceManager {
     String deviceId, {
     DeviceDiscoveryFilter? filter,
   }) async {
-    return devices.where((Device device) => device.id == deviceId).toList();
+    return devices.where((Device device) {
+      return device.id == deviceId || device.id.startsWith(deviceId);
+    }).toList();
```

Comment:
Did you mean to change this logic in this refactoring PR?

Reply:
Yes, it was needed for this test.

https://github.com/flutter/flutter/blob/8ee0f503ee6d4abae1968b10ebd122c55511dc2e/packages/flutter_tools/test/general.shard/runner/target_devices_test.dart#L155-L168

In the old version of this test, `findAllTargetDevices` basically mocked the returned device list so it always returned the device list given. So the mocked `getDevicesById` was never actually called.